nexus/db-queries/src/db/datastore/address_lot.rs

```diff
@@ -72,6 +93,38 @@ impl DataStore {
                     })
                     .collect();
 
+                let db_blocks: Vec<AddressLotBlock> =
+                    block_dsl::address_lot_block
+                        .filter(block_dsl::address_lot_id.eq(db_lot.id()))
+                        .filter(
+                            block_dsl::first_address.eq_any(
+                                blocks
+                                    .iter()
+                                    .map(|b| b.first_address)
+                                    .collect::<Vec<_>>(),
+                            ),
+                        )
+                        .filter(
+                            block_dsl::last_address.eq_any(
+                                blocks
+                                    .iter()
+                                    .map(|b| b.last_address)
+                                    .collect::<Vec<_>>(),
+                            ),
+                        )
+                        .get_results_async(&conn)
+                        .await?;
+
+                let blocks = blocks
+                    .into_iter()
+                    .filter(|b| {
+                        !db_blocks.iter().any(|db_b| {
+                            db_b.first_address == b.first_address
+                                || db_b.last_address == b.last_address
+                        })
+                    })
+                    .collect::<Vec<_>>();
```

Comment:
To be clear, this is looking for cases where:

- We have `blocks` that come from params
- In the database, we looked up `db_blocks`, which come from the same address lot and happen to have the same "first address" or the same "last address" as the parameter-supplied `blocks` (lines 99-114)

And this particular filter identifies:
- If there are **any** `blocks` where `db_blocks` has the same "first address" or "last address", we filter them out of `blocks`?

Am I misunderstanding this? Aren't we explicitly looking for that overlap - why then filter it out?

Reply:
We're _supposed_ to be filtering for blocks to insert, but I may have gotten my wires crossed here. I'll take a second look here.

Reply:
Yeah, this actually doesn't work quite right for what we're planning to return below. I'll need to tweak it. It should be _inserting_ the correct stuff though.

Reply:
@smklein this should be fixed now